system_metrics_collector/src/system_metrics_collector/main.cpp

```diff
@@ -67,12 +76,22 @@ int main(int argc, char ** argv)
     RCUTILS_LOG_ERROR_NAMED("main", "Unable to set debug logging for the memory node");
   }
 
+  r = rcutils_logging_set_logger_level(process_mem_node->get_name(), RCUTILS_LOG_SEVERITY_DEBUG);
+
+  if (r != 0) {
+    RCUTILS_LOG_ERROR_NAMED("main", "Unable to set debug logging for the process memory node");
+  }
+
   ex.add_node(cpu_node);
   ex.add_node(mem_node);
+  ex.add_node(process_mem_node);
   ex.spin();
 
   rclcpp::shutdown();
+
   cpu_node->stop();
   mem_node->stop();
+  process_mem_node->stop();
```

Comment:
You may want to write that into a try...catch for exception safety:

```c++
try {
  // do stuff
}
catch(...) {
  cleanUp();
  throw;
}
cleanUp();
```

Reply:
I'm not sure we need it right now because stop / setup stop does not currently throw. If we had throwing operations I would agree.